### DISCUSSION TITLE
In Encrypted database, How to check whether the db connection is succesful or not ?

### DISCUSSION THREAD

**Discussion by rajagopalx** [simolus3/drift#1502] [Q&A] [answered]
Hi,

I am using the encrypted database. If I input the wrong passphrase how to catch the exception

Currently, I am getting exception if I am inserting records into the db.
`[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: SqliteException(26): file is not a database, file is not a database (code 26)`

But, Is there any other way to check whether the database connection is success or not.

P.S: I am new to flutter development.

My Code
```
LazyDatabase _openConnection() {
  return LazyDatabase(() async {
    final dbFolder = await getApplicationDocumentsDirectory();
    final file = File(p.join(dbFolder.path, 'db.sqlite'));
    setupSqlCipher();
    return NativeDatabase(file, setup: (rawDb) {
      rawDb.execute("PRAGMA key = 'mypassword';");
    });
  });
}

@DriftDatabase(tables: [
  mytables,
], daos: [
])
class MyDatabase extends _$MyDatabase {
  MyDatabase() : super(_openConnection());
  @override
  int get schemaVersion => 1;
}

```
https://drift.simonbinder.eu/docs/other-engines/encryption/

**Comment by simolus3** [comment, marked as answer]
You could add a simple statement accessing the database in the `setup` callback, like this:

```dart
setup: (rawDb) {
  rawDb.execute("PRAGMA key = 'wrongpassword';");
  
  try {
    rawDb.execute('PRAGMA user_version;');
  } on SqliteException catch (e) {
    if (e.resultCode == 26) {
      // not a database, the password is probably wrong
    }
  }
},
```

**Reply by rajagopalx** [reply]
Thank You so much
